zerver/lib/transfer.py

```diff
@@ -0,0 +1,90 @@
+import os
+import logging
+
+from django.conf import settings
+from django.db import connection
+from mimetypes import guess_type
+
+from zerver.models import UserProfile, Attachment, RealmEmoji
+from zerver.lib.avatar_hash import user_avatar_path
+from zerver.lib.upload import read_local_file, S3UploadBackend, upload_image_to_s3
+from zerver.lib.parallel import run_parallel
+
+s3backend = S3UploadBackend()
+
+def transfer_uploads_to_s3(processes: int) -> None:
+    transfer_avatars_to_s3(processes)
+    transfer_message_files_to_s3(processes)
+    transfer_emoji_to_s3(processes)
+
+def transfer_avatars_to_s3(processes: int) -> None:
+    def _transfer_avatar_to_s3(user: UserProfile) -> int:
+        avatar_path = user_avatar_path(user)
+        file_path = os.path.join(settings.LOCAL_UPLOADS_DIR, "avatars", avatar_path) + ".original"
+        try:
+            with open(file_path, 'rb') as f:
+                s3backend.upload_avatar_image(f, user, user)
+                logging.info("Uploaded avatar for {} in realm {}".format(user.email, user.realm.name))
+        except FileNotFoundError:
+            pass
+        return 0
+
+    users = list(UserProfile.objects.all())
+    if processes == 1:
+        for user in users:
+            _transfer_avatar_to_s3(user)
+    else:  # nocoverage
+        output = []
+        connection.close()
+        for (status, job) in run_parallel(_transfer_avatar_to_s3, users, processes):
+            output.append(job)
```

Comment:
This isn't a blocker, but FYI, this is a really inefficient way to do it in parallel, because `run_parallel` forks a new process for each entry in the list.  

See `run_parallel_wrapper` for an example of distributing work so each thread does 1/n of the work and then exits, rather than having e.g. 1000 work items and 6 threads as this does.  I think probably we should do a refactor to extract that as reusable code and call a function of that form.